discovery-infra/test_infra/tools/static_network.py

```diff
@@ -0,0 +1,102 @@
+        tfvars = json.load(_file)
+
+    network_config = []
+
+    for count in range(tfvars['master_count']):
+        host_data = _prepare_host_static_network_data(tfvars['libvirt_master_macs'][count],
+                                                      tfvars['libvirt_secondary_master_macs'][count],
+                                                      tfvars['machine_cidr_addresses'],
+                                                      tfvars['provisioning_cidr_addresses'],
+                                                      count)
+        network_config.append(host_data)
+
+    for count in range(tfvars['worker_count']):
+        host_data = _prepare_host_static_network_data(tfvars['libvirt_worker_macs'][count], 
+                                                      tfvars['libvirt_secondary_worker_macs'][count],
+                                                      tfvars['machine_cidr_addresses'],
+                                                      tfvars['provisioning_cidr_addresses'],
+                                                      tfvars['master_count'] + count)
+        network_config.append(host_data)
+
+    return network_config
+
+
+def _prepare_host_static_network_data(primary_mac,
+                                      secondary_mac,
+                                      machine_cidrs,
+                                      provisioning_cidrs,
+                                      count):
+    interfaces = _prepare_interfaces(primary_mac, secondary_mac, machine_cidrs, provisioning_cidrs, count)
+    dns_resolver = _prepare_dns_resolver(machine_cidrs)
+    routes = _prepare_routes(primary_mac, machine_cidrs)
+
+    host_network_config = {'interfaces': interfaces,
+                           'dns-resolver': dns_resolver,
+                           'routes': routes}
+
+    return yaml.dump(host_network_config)
+
+
+def _prepare_interfaces(primary_mac, secondary_mac, machine_cidrs, provisioning_cidrs, count):
+    primary_interface = _prepare_interface(primary_mac, machine_cidrs, count)
+    secondary_interface = _prepare_interface(secondary_mac, provisioning_cidrs, count)
+    return [primary_interface, secondary_interface]
+
+def _prepare_interface(mac, cidrs, host_count):
+    ipv4_dict = {}
+    ipv6_dict = {}
+
+    ipv6_version, ip_dict = _prepare_ip_dict(cidrs[0], host_count)
+
+    if not ipv6_version:
+        ipv4_dict = ip_dict
+    else:
+        ipv6_dict = ip_dict
+
+    if not ipv6_version and len(cidrs) > 1:
+        _, ipv6_dict = _prepare_ip_dict(cidrs[1], host_count)
+
+    interface = {'name': mac.replace(':', ''),
+                 'type': 'ethernet',
+                 'state': 'up'}
+    if ipv4_dict:
+        interface['ipv4'] = ipv4_dict
+    if ipv6_dict:
+        interface['ipv6'] = ipv6_dict
+
+    return interface
+
+
+def _prepare_ip_dict(cidr, host_count):
+    network = ip_network(cidr)
+    ip = str(ip_address(network.network_address) + 30 + host_count)
```

Comment:
pls consider storing the value '30' in a variable to describe its role (e.g. ip_address_offset)